lesson_09/oop/oop_app/src/test/java/com/codedifferently/lesson9/sherllin/BookTest.java

```diff
@@ -0,0 +1,123 @@
+// Allows us to return books from patrons.
+
+public class BookTest {
+
+  @Test
+  public void testBookProperties() {
+    // Arrange
+    String title = "The Alchemist";
+    long isbn = 9780061122415L;
+    String author = "Paulo Coelho";
+    int pages = 208;
+    boolean checkedOut = false;
+
+    // Act
+    Book theAlchemist = new Book(title, isbn, author, pages, checkedOut);
+
+    // Assert
+    assertEquals(title, theAlchemist.getBookTitle());
+    assertEquals(isbn, theAlchemist.getIsbn());
+    assertEquals(author, theAlchemist.getAuthor());
+    assertEquals(pages, theAlchemist.getPages());
+    assertEquals(checkedOut, theAlchemist.checkedOut());
+
+    // Test additional books
+    Book theCatInTheHat = new Book(title, isbn, author, pages, checkedOut);
+    assertEquals(title, theCatInTheHat.getBookTitle());
+    assertEquals(isbn, theCatInTheHat.getIsbn());
+    assertEquals(author, theCatInTheHat.getAuthor());
+    assertEquals(pages, theCatInTheHat.getPages());
+    assertEquals(checkedOut, theCatInTheHat.checkedOut());
+
+    Book atomicHabits = new Book(title, isbn, author, pages, checkedOut);
+    assertEquals(title, atomicHabits.getBookTitle());
+    assertEquals(isbn, atomicHabits.getIsbn());
+    assertEquals(author, atomicHabits.getAuthor());
+    assertEquals(pages, atomicHabits.getPages());
+    assertEquals(checkedOut, atomicHabits.checkedOut());
+  }
+
+  @Test
+  public void testPatronProperties() {
+    Patron patron = new Patron("Nathalie", 3);
+    String name = "Nathalie";
+    int bookCheckedOut = 3;
+    assertEquals(name, patron.getName());
+    assertEquals(bookCheckedOut, patron.getBookCheckedOut());
+  }
+
+  @Test
+  public void testAddBook() {
+    Library library = new Library();
+    library.addBook(9780316769488L);
+    library.addBook(9780142437209L);
+    library.addBook(9780743273565L);
+
+    assertEquals(3, library.getBookStatus().size());
+    assertEquals(Boolean.FALSE, library.getBookStatus().get(9780316769488L));
+    assertEquals(Boolean.FALSE, library.getBookStatus().get(9780142437209L));
+    assertEquals(Boolean.FALSE, library.getBookStatus().get(9780743273565L));
+  }
+
+  @Test
+  public void testRemoveBook() {
+    Library library = new Library();
+    library.addBook(9780316769488L);
+    library.addBook(9780142437209L);
+    library.removeBook(9780316769488L);
+
+    Map<Long, Boolean> expectedBookStatus = new HashMap<>();
+    expectedBookStatus.put(9780142437209L, false);
+
+    assertEquals(expectedBookStatus, library.getBookStatus());
+  }
+
+  @Test
+  public void testRegisterPatron() {
+    Library library = new Library();
+    library.registerPatron("Lily");
+    library.registerPatron("Dali");
+
+    assertEquals(2, library.getPatronBooks().size());
+    assertEquals(Integer.valueOf(0), library.getPatronBooks().get("Lily"));
+    assertEquals(Integer.valueOf(0), library.getPatronBooks().get("Dali"));
+  }
+
+  @Test
+  public void testCheckoutAndReturnBook() {
```

Comment:
This should be two tests, one for testing checking out a book and the other for returning.